### PR TITLE
fix: add null/undefined check to attribution data

### PR DESCRIPTION
**Pull request by alanjcharles**
- adds safety check for attribution values in `Install Attributed` events in Braze Plugin

### CODE REVIEW

File: packages/plugins/plugin-braze/src/methods/track.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@ export default (payload: TrackEventType) => {
   if (payload.event === 'Install Attributed') {
     if (payload.properties?.campaign) {
       const attributionData: any = payload.properties.campaign;
+      for (const [attribution, data] of Object.entries(attributionData)) {
+        if (data === undefined || data === null) {
+          attributionData[attribution] = '';
+        }
+      }
```
Comment (oscb) [inline packages/plugins/plugin-braze/src/methods/track.ts:12]: I don't think this code will handle `undefined` correctly because `Object.entries` will not return the key for those elements (they are essentially non-existent)

Maybe try a different approach using null-coalescing on the properties down below and having an constant object with initial values for each of them so that we can tweak them individually if needed.

Also add a test for both `null` and `undefined` scenarios! (specially the kind of `undefined` where you don't explicitely set it as `undefined`)

**Review by oscb — CHANGES_REQUESTED** [review]
Left a single comment about an edge case this won't cover. It would be good if you can add tests for this as well!

**Review by oscb — APPROVED** [review]
lgtm!

**Comment by alanjcharles** [comment]
#677
